docs/topics/signals.txt

```diff
@@ -11,38 +11,34 @@ allow certain *senders* to notify a set of *receivers* that some action has
 taken place. They're especially useful when many pieces of code may be
 interested in the same events.
 
-Django provides a :doc:`set of built-in signals </ref/signals>` that let user
-code get notified by Django itself of certain actions. These include some useful
-notifications:
+For example, a third-party app can register to be notified of settings
+changes::
 
-* :data:`django.db.models.signals.pre_save` &
-  :data:`django.db.models.signals.post_save`
+    from django.apps import AppConfig
+    from django.core.signals import setting_changed
 
-  Sent before or after a model's :meth:`~django.db.models.Model.save` method
-  is called.
-
-* :data:`django.db.models.signals.pre_delete` &
-  :data:`django.db.models.signals.post_delete`
-
-  Sent before or after a model's :meth:`~django.db.models.Model.delete`
-  method or queryset's :meth:`~django.db.models.query.QuerySet.delete`
-  method is called.
+    def my_callback(sender, **kwargs):
+        print("Setting changed!")
 
-* :data:`django.db.models.signals.m2m_changed`
+    class MyAppConfig(AppConfig):
+        ...
 
-  Sent when a :class:`~django.db.models.ManyToManyField` on a model is changed.
+        def ready(self):
+            setting_changed.connect(my_callback)
 
-* :data:`django.core.signals.request_started` &
-  :data:`django.core.signals.request_finished`
+Django's :doc:`built-in signals </ref/signals>` let user code get notified of
+certain actions.
 
-  Sent when Django starts or finishes an HTTP request.
+You can also define and send your own custom signals. See
+:ref:`defining-and-sending-signals` below.
 
-See the :doc:`built-in signal documentation </ref/signals>` for a complete list,
-and a complete explanation of each signal.
+.. warning::
 
-You can also `define and send your own custom signals`_; see below.
+    Signals give the appearance of loose-coupling, but they can quickly lead to
```

Comment:
No hyphen required

also I would argue they do provide loose(r) coupling, not just the appearance thereof?

```suggestion
    Signals give the appearance of loose coupling, but they can quickly lead to
```

Reply:
I don't think the code is really decoupled 😅 Rather it's just **somewhere else**. Not the same thing. 😜

Reply:
They result in _apparent_ loose coupling. In my opinion, the resulting code is tightly coupled. But usually located in another module where I wouldn't intuitively look for it.

"Loose coupling" [definition](https://en.wikipedia.org/wiki/Loose_coupling):

> In computing and systems design, a loosely coupled system is one in which components are weakly associated with each other, and thus changes in one component least affect existence or performance of another component

"Coupling" [definition](https://en.wikipedia.org/wiki/Coupling_(computer_programming)):

> coupling is the degree of interdependence between software modules; a measure of how closely connected two routines or modules are

I believe that a full model definition includes the signals defined for it. Or that's how I would like to think about it. Signals make that harder. As opposed to the alternatives suggested (as manager functions, overriding instance functions).

That's why they are coupled but give the impression of loose coupling.

Reply:
Thanks for the feedback @adamchainz ! I added commits with both of your suggestions.
And unresolved this thread, so we can agree on it.

Reply:
Fair enough, I don’t want to split hairs on this! Good work all!